src/NuGet.Clients/NuGet.PackageManagement.PowerShellCmdlets/PackageActionBaseCommand.cs

```diff
@@ -293,8 +293,23 @@ protected override void EndProcessing()
 
         protected async Task CheckPackageManagementFormat()
         {
-            // check if Project has any packages installed
-            if (!(await Project.GetInstalledPackagesAsync(Token)).Any())
+            bool packagesConfigAndSupportsPackageReferences = false;
+
+            if (Project.ProjectStyle == ProjectModel.ProjectStyle.PackagesConfig)
+            {
+                try
+                {
+                    packagesConfigAndSupportsPackageReferences = Project.ProjectServices.Capabilities.SupportsPackageReferences;
```

Comment:
Why not use the exact same check as @kartheekp-ms recommended?
This won't throw for PC projects.

The logic is inconsistently applied, so the whole thing requires some refactoring.